Hippo.Web/ClientApp/src/components/Order/BillingsFields.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useState } from "react";
+import { useFieldArray, useFormContext, useWatch } from "react-hook-form";
+import { OrderModel } from "../../types";
+
+import FormField from "../../Shared/Form/FormField";
+import HipButton from "../../Shared/HipButton";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPlus, faTrash, faSearch } from "@fortawesome/free-solid-svg-icons";
+import { authenticatedFetch } from "../../util/api";
+import ChartStringValidation from "./ChartStringValidation";
+import OrderFormField from "./OrderFormField";
+import { Card, CardBody, CardTitle } from "reactstrap";
+
+declare const window: Window &
+  typeof globalThis & {
+    Finjector: any;
+  };
+
+type BillingsFieldsProps = {
+  readOnly: boolean;
+};
+
+const BillingsFields: React.FC<BillingsFieldsProps> = ({ readOnly }) => {
+  const {
+    control,
+    register,
+    getValues,
+    setValue,
+    formState: { errors },
+  } = useFormContext<OrderModel>();
+
+  const { fields, append, remove, update } = useFieldArray({
+    control,
+    name: "billings",
+  });
+
+  const addBilling = () => {
+    const percentTotal = getValues("percentTotal");
+    let percentToSet = 100 - percentTotal;
+    if (percentToSet < 0) {
+      percentToSet = 0;
+    }
+    if (percentToSet > 100) {
+      percentToSet = 100;
+    }
+
+    append({
+      id: 0,
+      chartString: "",
+      percentage: percentToSet.toString(),
+      chartStringValidation: {
+        isValid: true,
+        description: "",
+        accountManager: "",
+        accountManagerEmail: "",
+        message: "",
+        warning: "",
+      },
+    });
+  };
+
+  const removeBilling = (index: number) => {
+    remove(index);
+  };
+
+  const [notification, setNotification] = useState<string>("");
+  const showNotification = (message: string) => {
+    setNotification(message);
+    // Optionally, clear the notification after some time
+    setTimeout(() => setNotification(""), 5000);
```

Comment:
this will get squirrely if this is called >1, since the old setTimeout will still be running and can clear the new notification instantly.  better to either just use toast of if you want to use local, then handle the clearTimeout() logic properly to restart the timeout whenever a new notification is set

Reply:
@srkirkland Not sure how to get toasts to work with that. But... it works fine if it is called quickly multiple times.

Reply:
@jSylvestre i think he just means `  const [notification, setNotification] = usePromiseNotification();` instead of a string

Reply:
I don't see how to get that to work. I only want to show a message for a short time if I've changed the natural account. Not every time validate the chart string is called.